qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -215,6 +209,35 @@ def _on_config_changed(self, option):
             self.maybe_hide()
         elif option == 'statusbar.padding':
             self._set_hbox_padding()
+        elif option == 'statusbar.widgets':
+            self._draw_widgets()
+
+    def _draw_widgets(self):
+        """Draw statusbar widgets."""
+        # Start with widgets hidden and show them when needed
+        for widget in [self.url, self.percentage,
+                       self.backforward, self.tabindex]:
+            self._hbox.removeWidget(widget)
+            widget.hide()
```

Comment:
I think it'd be better to first hide the widget and then remove it from the hbox. I can't actually reproduce it right now, but I think when removing a widget which is shown, it might briefly be shown as a separate window before being hidden.

Reply:
Ok, I switched it.